Source/WebCore/page/Navigator.cpp

```diff
@@ -137,15 +137,25 @@ bool Navigator::canShare(Document& document, const ShareData& data)
     if (!document.isFullyActive() || !validateWebSharePolicy(document))
         return false;
 
-    bool hasShareableTitleOrText = !data.title.isNull() || !data.text.isNull();
-    bool hasShareableURL = !!shareableURLForShareData(document, data);
 #if ENABLE(FILE_SHARE)
     bool hasShareableFiles = document.settings().webShareFileAPIEnabled() && !data.files.isEmpty();
 #else
     bool hasShareableFiles = false;
 #endif
 
-    return hasShareableTitleOrText || hasShareableURL || hasShareableFiles;
+    if (data.title.isNull() && data.text.isNull() && data.url.isNull() && !hasShareableFiles)
+        return false;
+
+#if ENABLE(FILE_SHARE)
+    bool hasTitleTextOrURL = !data.title.isNull() || !data.text.isNull() || !data.url.isNull();
+    if (!hasTitleTextOrURL && !hasShareableFiles)
+        return false;
+#endif
+
+    if (!data.url.isNull() && !shareableURLForShareData(document, data))
+        return false;
+
+    return true;
```

Comment:
Could be a one liner:
return data.url.isNull() || shareableURLForShareData(document, data);

Reply:
As above. Will fix.